superset/assets/src/dashboard/reducers/getInitialState.js

```diff
@@ -173,22 +181,50 @@ export default function(bootstrapData) {
           });
         }
 
+        // backward compatible:
+        // merge scoped filter settings with old global immune settings
+        const scopesByChartId = Object.keys(columns).reduce((map, column) => {
+          const scopeSettings = {
+            ...filterScopes[key],
+          };
+          const { scope, immune } = {
+            ...DASHBOARD_FILTER_SCOPE_GLOBAL,
+            ...scopeSettings[column],
+          };
+          const immuneChartIds = new Set(filterImmuneSlices.slice());
+          if (!isEmpty(filterImmuneSliceFields)) {
```

Comment:
Why do we need to check for emptiness? If empty `.forEach` should not do anything already.

Reply:
fixed.